### PR TITLE
Refine Pyramidbox infer and Pyramidbox configure

### LINKED ISSUE #993: Add Pyramidbox inference configure.
预测配置需要crop操作：

1. 前面的VGG配置如下：

    conv1 (C, H1, W1) ->pooling (C, H2, W2) (stride是2) ->conv2(C, H2, W2)
    H1/W1 是偶数时， `H1=2 * H2`, `W1 = 2 * W2`。

2. FPN部分：

    conv2(C, H2, W2)->de-conv1(C, H3, W3)，
    de-conv这里上采样，是将feature map扩大2倍。H3 = 2 * H2， W3 = 2 * W2。
    de-conv1 * conv1 作为输入

但是：如果H1,W1是奇数，1.中的pooling会取整(假定设置上取整)，例如 W1=59，则，W2=30.
则 W3 = 60，此时， de-conv1和conv1的shape不一样，无法做点对点相乘。因为需要将de-conv1做crop。

### PR DESCRIPTION
**Pull request by baiyfbupt**
-  Refine infer.py to use pyramid infer entire val set.
Fix https://github.com/PaddlePaddle/models/issues/995
-  Refine Pyramidbox configure for inference by @qingqing01 
Fix #993

### CODE REVIEW

File: fluid/face_detection/infer.py [hunk 1]
```diff
@@ -28,65 +29,282 @@
...
 
-    network = PyramidBox(
-        image_shape,
-        num_classes,
-        sub_network=args.use_pyramidbox,
-        is_infer=True)
-    infer_program, nmsed_out = network.infer()
-    fetches = [nmsed_out]
 
+def detect_face(image, image_shape, raw_image, shrink):
```
Comment (qingqing01) [inline fluid/face_detection/infer.py]: image_shape can get from image ?

File: fluid/face_detection/infer.py [hunk 2]
```diff
@@ -28,65 +29,282 @@
...
+        img = np.swapaxes(img, 1, 2)
+        img = np.swapaxes(img, 1, 0)
+    # RBG to BGR
+    img = img[[2, 1, 0], :, :]
+    img = img.astype('float32')
+    img -= np.array(
+        [104., 117., 123.])[:, np.newaxis, np.newaxis].astype('float32')
+    img = img * 0.007843
+    img = [img]
+    img = np.array(img)
```
Comment (qingqing01) [inline fluid/face_detection/infer.py:127]: Put above image preprocess to another function.

File: fluid/face_detection/infer.py [hunk 3]
```diff
@@ -28,65 +29,282 @@
...
+                fetches = [nmsed_out]
+                feeder = fluid.DataFeeder(
+                    place=place, feed_list=network.feeds())
+
+                fluid.io.load_persistables(
+                    exe, args.model_dir, main_program=model_program)
+                #fluid.io.load_vars(exe, args.model_dir, predicate=if_exist)
+
+                detection, = exe.run(infer_program,
+                                     feed=feeder.feed([img]),
```
Comment (qingqing01) [inline fluid/face_detection/infer.py]:
```
feed={'image': img}
```

File: fluid/face_detection/infer.py [hunk 4]
```diff
@@ -28,65 +29,282 @@
...
+                                     fetch_list=fetches,
+                                     return_numpy=False)
+                detection = np.array(detection)
```
Comment (qingqing01) [inline fluid/face_detection/infer.py]: Remove line 170 and line 171.

File: fluid/face_detection/infer.py [hunk 5]
```diff
@@ -28,65 +29,282 @@
...
+    img = img[[2, 1, 0], :, :]
+    img = img.astype('float32')
+    img -= np.array(
+        [104., 117., 123.])[:, np.newaxis, np.newaxis].astype('float32')
+    img = img * 0.007843
+    img = [img]
+    img = np.array(img)
+
+    scope = fluid.core.Scope()
+    model_program = fluid.Program()
```
Comment (qingqing01) [inline fluid/face_detection/infer.py]: model_program -> main_program

File: fluid/face_detection/infer.py [hunk 6]
```diff
@@ -28,65 +29,282 @@
...
+    with fluid.scope_guard(scope):
+        with fluid.unique_name.guard():
+            with fluid.program_guard(model_program, startup_program):
+                fetches = []
+                network = PyramidBox(
+                    image_shape,
+                    num_classes,
+                    sub_network=args.use_pyramidbox,
+                    is_infer=True)
+                infer_program, nmsed_out = network.infer()
```
Comment (qingqing01) [inline fluid/face_detection/infer.py]:
```
infer_program, nmsed_out = network.infer(main_program)
```

File: fluid/face_detection/infer.py (hunk 2 above)
Comment (baiyfbupt) [reply fluid/face_detection/infer.py:127]: Done

File: fluid/face_detection/infer.py (hunk 3 above)
Comment (baiyfbupt) [reply fluid/face_detection/infer.py]: Done

File: fluid/face_detection/infer.py (hunk 5 above)
Comment (baiyfbupt) [reply fluid/face_detection/infer.py]: Done

File: fluid/face_detection/infer.py (hunk 6 above)
Comment (baiyfbupt) [reply fluid/face_detection/infer.py]: Done

File: fluid/face_detection/infer.py (hunk 1 above)
Comment (baiyfbupt) [reply fluid/face_detection/infer.py]: Done

**Review by qingqing01 — APPROVED** [review]
Approved.   Only build Program once will add in the future.

**Pull request merged**